airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -214,19 +214,22 @@ public static void main(String[] args) throws Exception {
 
     Optional<String> airbyteDatabaseVersion = jobPersistence.getVersion();
     boolean isKubernetes = configs.getWorkerEnvironment() == WorkerEnvironment.KUBERNETES;
-    if (!isKubernetes && airbyteDatabaseVersion.isPresent() && !AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())
+    if (airbyteDatabaseVersion.isPresent() && !AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())
         && !isDatabaseVersionAheadOfAppVersion(airbyteVersion, airbyteDatabaseVersion.get())) {
-      runAutomaticMigration(configRepository, jobPersistence, airbyteVersion, airbyteDatabaseVersion.get());
-      // After migration, upgrade the DB version
-      airbyteDatabaseVersion = jobPersistence.getVersion();
+      if (!isKubernetes || new AirbyteVersion(airbyteDatabaseVersion.get()).patchVersionCompareTo(KUBE_SUPPORT_FOR_AUTOMATIC_MIGRATION) >= 0) {
```

Comment:
suggestion: simplify this conditional check.
```suggestion
      var isDocker = !isKubernetes;
      var versionSupportsAutoMigrate =  new AirbyteVersion(airbyteDatabaseVersion.get()).patchVersionCompareTo(KUBE_SUPPORT_FOR_AUTOMATIC_MIGRATION) >= 0;
      if (!isKubernetes || afterAutoMigrateVersion) {
```